conjure/objectwriter.go

```diff
@@ -34,6 +34,9 @@ func writeObjectType(file *jen.Group, objectDef *types.ObjectType) {
 	file.Add(objectDef.Docs.CommentLine()).Type().Id(objectDef.Name).StructFunc(func(structDecl *jen.Group) {
 		for _, fieldDef := range objectDef.Fields {
 			fieldName := fieldDef.Name
+			if _, isOptional := fieldDef.Type.(*types.Optional); isOptional {
+				fieldName += ",omitempty"
```

Comment:
nit: let's rename `fieldName`, this is no longer only representing only a field name.